akka-http-core/src/test/scala/akka/http/impl/engine/client/HostConnectionPoolSpec.scala

```diff
@@ -295,6 +297,19 @@ class HostConnectionPoolSpec extends AkkaSpec(
 
         val conn2 = expectNextConnection()
       }
+      "don't buffer an unreasonable number of outgoing responses" in new SetupWithServerProbes(_.withMaxConnections(1).withMinConnections(1)) {
+        val conn1 = expectNextConnection()
+
+        def oneCycle(): Unit = {
+          pushRequest()
+          conn1.expectRequest()
+          conn1.pushResponse()
+        }
+
+        eventually {
+          a[Throwable] should be thrownBy oneCycle()
+        }
```

Comment:
what's the error message here?

Reply:
A timeout that a push / expect didn't go through because of backpressure. Hard to test otherwise.

Reply:
Testing method is fine, just wondering if the message is understandable :)

Reply:
Added a comment.